app/services/plans/update_service.rb

```diff
@@ -87,6 +89,20 @@ def bill_charges_monthly?
       params[:bill_charges_monthly] || false
     end
 
+    def cascade_subscription_fee_update(old_amount_cents)
+      return unless cascade?
+      return if plan.children.empty?
+      return if old_amount_cents == plan.amount_cents
```

Comment:
Maybe we could move this check as the first one in the method as it would avoid calling the db for the `plan.children.empty?` if no update is required

Reply:
Make sense, moved up!